Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -33,10 +33,10 @@ export class azureclitask {
             else {
                 var script: string = tl.getInput("inlineScript", true);
                 if (os.type() != "Windows_NT") {
-                    scriptPath = path.join(os.tmpdir(), "azureclitaskscript" + new Date().getTime() + ".sh");
+                    scriptPath = path.join(tl.getVariable('Agent.TempDirectory'), "azureclitaskscript" + new Date().getTime() + ".sh");
```

Comment:
why does this need to change. Also it seems that below you are handling when this variable is not available; but you are not doing that here

Reply:
done